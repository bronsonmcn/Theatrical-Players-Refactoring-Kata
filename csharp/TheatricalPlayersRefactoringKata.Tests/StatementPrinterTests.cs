using System;
using System.Collections.Generic;
using ApprovalTests;
using ApprovalTests.Reporters;
using Xunit;

namespace TheatricalPlayersRefactoringKata.Tests
{
    public class StatementPrinterTests
    {
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void test_statement_example()
        {
            var plays = new Dictionary<string, Play>();
            plays.Add("hamlet", new Play("Hamlet", PlayType.Tragedy));
            plays.Add("as-like", new Play("As You Like It", PlayType.Comedy));
            plays.Add("othello", new Play("Othello", PlayType.Tragedy));

            Invoice invoice = new Invoice("BigCo", new List<Performance>{new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)});
            
            StatementPrinter statementPrinter = new StatementPrinter();
            var result = statementPrinter.Print(invoice, plays, PrintType.String);

            Approvals.Verify(result);
        }
        
        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void test_statement_example_html()
        {
            var plays = new Dictionary<string, Play>();
            plays.Add("hamlet", new Play("Hamlet", PlayType.Tragedy));
            plays.Add("as-like", new Play("As You Like It", PlayType.Comedy));
            plays.Add("othello", new Play("Othello", PlayType.Tragedy));

            Invoice invoice = new Invoice("BigCo", new List<Performance>{new Performance("hamlet", 55),
                new Performance("as-like", 35),
                new Performance("othello", 40)});
            
            StatementPrinter statementPrinter = new StatementPrinter();
            var result = statementPrinter.Print(invoice, plays, PrintType.Html);

            Approvals.Verify(result);
        }
        
        [Fact]
        public void test_statement_with_new_play_types()
        {
            var plays = new Dictionary<string, Play>();
            plays.Add("henry-v", new Play("Henry V", PlayType.History));
            plays.Add("as-like", new Play("As You Like It", PlayType.Pastoral));

            Invoice invoice = new Invoice("BigCoII", new List<Performance>{new Performance("henry-v", 53),
                new Performance("as-like", 55)});
            
            StatementPrinter statementPrinter = new StatementPrinter();

            Assert.Throws<Exception>(() => statementPrinter.Print(invoice, plays, PrintType.String));
        }
    }
}
