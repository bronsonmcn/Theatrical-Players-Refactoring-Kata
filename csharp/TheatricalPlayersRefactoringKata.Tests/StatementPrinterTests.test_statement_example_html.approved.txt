<h1>Statement for BigCo</h1>
<p>  Hamlet: $650.00 (55 seats)</p>
<p>  As You Like It: $580.00 (35 seats)</p>
<p>  Othello: $500.00 (40 seats)</p>
<b>Amount owed is $1,730.00</b>
<b>You earned 47 credits</b>
